config/jobs/kubernetes/sig-network/sig-network-kind.yaml

```diff
@@ -100,6 +100,52 @@ presubmits:
       testgrid-tab-name: pr-sig-network-kind, ipvs, dual
       description: Runs tests against a Dual Stack Kubernetes in Docker cluster
       testgrid-alert-email: antonio.ojea.garcia@gmail.com
+  - name: pull-kubernetes-e2e-kind-load-balancers
+    cluster: k8s-infra-prow-build
+    optional: true
+    always_run: false
+    skip_report: false
+    run_if_changed: '^(pkg\/proxy)'
+    decorate: true
+    labels:
+      preset-dind-enabled: "true"
+      preset-kind-volume-mounts: "true"
+    decoration_config:
+      timeout: 60m
+      grace_period: 15m
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/krte:v20240111-cf1d81388e-master
+        command:
+        - wrapper.sh
+        - bash
+        - -c
+        - curl -Lo ./cloud-provider-kind https://github.com/kubernetes-sigs/cloud-provider-kind/releases/download/v0.0.1/cloud-provider-kind-linux-amd64
+        - chmod +x cloud-provider-kind && nohup cloud-provider-kind > ccm-kind.log 2>&1 &
+        - curl -sSL https://kind.sigs.k8s.io/dl/latest/linux-amd64.tgz | tar xvfz - -C "${PATH%%:*}/" && e2e-k8s.sh
```

Comment:
something needs to pass `--cloud-provider=external` to kubelet, right?

Also, should we just add the cloud provider to all of the kind jobs?

Reply:
oh right, another experimental job :/

Reply:
> Also, should we just add the cloud provider to all of the kind jobs?

not really, the goal is to leave the kubernetes code providerless ,so for the core part it should not be necessary, there are still things we need to test, but does not look they should be in the critical path of the presubmits.

For these things we always use the periodics, because is not common to break and if it breaks is easy to spot the change that caused the issue and revert it

Reply:
ok, now I remember the script that runs th tests is here https://github.com/kubernetes-sigs/cloud-provider-kind/blob/main/hack/ci/e2e.sh

Reply:
updated

Reply:
> > Also, should we just add the cloud provider to all of the kind jobs?
> 
> not really, the goal is to leave the kubernetes code providerless ,so for the core part it should not be necessary, there are still things we need to test, but does not look they should be in the critical path of the presubmits.

Hm... ok, but it seems like it would be good to add this to the ipvs and nftables jobs wouldn't it? Because currently there is no job that would catch LoadBalancer bugs in ipvs or nftables...

Reply:
you are right,  we are also missing the coverage on the service cloud provider code https://github.com/kubernetes/kubernetes/pull/121091  ... and all the external cloud provider code in general ... the challenge is that we need to revendor the code, let me rework this to cover all these things